modules/geolocation/geolocation.py

```diff
@@ -129,3 +130,71 @@ def calculate_pixel_to_geo_mapping(self):
 
         # Return matrix product of mappedGeoMatrix and sourcePixelMatrixInverse
         return (mappedGeoMatrix.dot(sourcePixelMatrixInverse))
+    
+class OutputAnalysis:
+
+    def get_best_location(self,inputLocationTupleList):
+
+    # Splits the 3D numpy arrray into three separate arrays
+
+        try:
+            coordPair = np.vstack(inputLocationTupleList[:, 0]).astype(np.float64)
+            errorArray = np.vstack(inputLocationTupleList[:, 1]).astype(np.float64)
+            confidenceArray = np.vstack(inputLocationTupleList[:, 2]).astype(np.float64)
+
+    # Splits coordinate pair into x-coord and y-coord to remove outliers
+            xCoord = np.vstack(coordPair[:, [0][0]]).astype(np.float64)
+            yCoord = np.vstack(coordPair[:, [1][0]]).astype(np.float64)
+            xCoordTrimmed=[]
+            yCoordTrimmed=[]
+            errorTrimmed=[]
+
+
+            xCoordMedian = np.median(xCoord)
+            yCoordMedian = np.median(yCoord)
+            errorMedian = np.median(errorArray)
+
+    # A mmodified version of the trimmed mean calculation. Here the distance measuremnt is
+    # distance between every x,y point from the median coordinate pair, x median and y median.
+
+            for (x,y) in zip(xCoord,yCoord):
+                distance = math.sqrt((pow((x-xCoordMedian),2)+(pow((y-yCoordMedian),2))))
+
+                if distance<3.5:
```

Comment:
Is there any relevant literature specifying a distance of 3.5 as the cut-off for an outlier? Is it based off the 
```
x is an outlier IFF x > Q3 * 1.5 * InterQuartileRange
```
formula? 

From an engineering perspective, if we are setting a magic constant like this, put it in a constant called ```OUTLIER_DISTANCE = 3.5``` at the top of the python file

Reply:
I had based the 3.5 off where I originally got the formula from, which recommended 3.5 as a threshold value for trimming using Z-scores, and what I had previously implemented was a modified version of Z-score.

Reply:
Are you using Z-scores right now? It doesn't look like it at the moment.

Reply:
Use 7 then. There's going to be 3 clinics close together and we don't currently have a way of distinguishing them.

Reply:
Z-scores could be used if you saved all of the distances, but I'm sure you want to work on other tasks and we need to get other tasks done.

Add a `# TODO` comment stating this.